tls/s2n_signature_algorithms.c

```diff
@@ -50,13 +52,23 @@ static int s2n_signature_scheme_valid_to_offer(struct s2n_connection *conn, cons
 static int s2n_signature_scheme_valid_to_accept(struct s2n_connection *conn, const struct s2n_signature_scheme *scheme)
 {
     POSIX_ENSURE_REF(scheme);
+    POSIX_ENSURE_REF(conn);
 
     POSIX_GUARD(s2n_signature_scheme_valid_to_offer(conn, scheme));
 
     if (scheme->maximum_protocol_version != S2N_UNKNOWN_PROTOCOL_VERSION) {
         POSIX_ENSURE_LTE(conn->actual_protocol_version, scheme->maximum_protocol_version);
     }
 
+    POSIX_ENSURE_NE(conn->actual_protocol_version, S2N_UNKNOWN_PROTOCOL_VERSION);
+
+    if (conn->actual_protocol_version >= S2N_TLS13) {
```

Comment:
Nit, but it'll be clearer that this ENSURE_NE is for these checks if you group them together. Otherwise, you'd probably want a comment on the ENSURE, as suggested by you feeling the need to leave a PR comment :) But with them grouped together, I don't think the comment is necessary.
```suggestion
    POSIX_ENSURE_NE(conn->actual_protocol_version, S2N_UNKNOWN_PROTOCOL_VERSION);
    if (conn->actual_protocol_version >= S2N_TLS13) {
```